src/integrations/release.ts

```diff
@@ -0,0 +1,66 @@
+import { addGlobalEventProcessor, getCurrentHub } from '@sentry/core';
+import type { Event, Integration } from '@sentry/types';
+
+import { NATIVE } from '../wrapper';
+
+/** Release integration responsible to load release from file. */
+export class Release implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Release';
+  /**
+   * @inheritDoc
+   */
+  public name: string = Release.id;
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(): void {
+    addGlobalEventProcessor(async (event: Event) => {
+      const self = getCurrentHub().getIntegration(Release);
+      if (!self) {
+        return event;
+      }
+
+      const options = getCurrentHub().getClient()?.getOptions();
+
+      /*
+        __sentry_release and __sentry_dist is set by the user with setRelease and setDist. If this is used then this is the strongest.
+        Otherwise we check for the release and dist in the options passed on init, as this is stronger than the release/dist from the native build.
+      */
+      if (typeof event.extra?.__sentry_release === 'string') {
+        event.release = `${event.extra.__sentry_release}`;
```

Comment:
M: `Sentry.setRelease()` is only in RN and will be removed in the next major. Let's read the release only from options.